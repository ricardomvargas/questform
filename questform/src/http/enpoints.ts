const SERVICE_PATH = 'http://localhost/questform-services/';

/** SURVEYS */
export const surveysGet = `${SERVICE_PATH}/questforms/get.php`;
export const surveysGetById = `${SERVICE_PATH}/questforms/getbyid.php`;
export const surveysNew = `${SERVICE_PATH}/questforms/new.php`;
export const surveysUpdate = `${SERVICE_PATH}/questforms/update.php`;
export const surveysDelete = `${SERVICE_PATH}/questforms/delete.php`;
/** QUESTIONS */
export const questionsGet = `${SERVICE_PATH}/questions/get.php`;
export const questionsGetById = `${SERVICE_PATH}/questions/getbyid.php`;
export const questionsNew = `${SERVICE_PATH}/questions/new.php`;
export const questionsUpdate = `${SERVICE_PATH}/questions/update.php`;
export const questionsDelete = `${SERVICE_PATH}/questions/delete.php`;
/** ANSWERS */
export const answersGet = `${SERVICE_PATH}/answers/get.php`;
export const answersGetById = `${SERVICE_PATH}/answers/getbyid.php`;
export const answersNew = `${SERVICE_PATH}/answers/new.php`;
export const answersUpdate = `${SERVICE_PATH}/answers/update.php`;
/** OPTIONS */
export const optionsGet = `${SERVICE_PATH}/options/get.php`;
export const optionsGetById = `${SERVICE_PATH}/options/getbyid.php`;
export const optionsNew = `${SERVICE_PATH}/options/new.php`;
