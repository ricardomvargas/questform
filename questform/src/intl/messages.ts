import appMessagesEN from './app/en';
import appMessagesPT from './app/pt';
import dashboardEN from './pages/dashboard/en';
import dashboardPT from './pages/dashboard/pt';
import questionEN from './pages/question/en';
import questionPT from './pages/question/pt';
import surveyEN from './pages/survey/en';
import surveyPT from './pages/survey/pt';

export default {
  en: {
    ...appMessagesEN,
    ...dashboardEN,
    ...questionEN,
    ...surveyEN,
  },
  pt: {
    ...appMessagesPT,
    ...dashboardPT,
    ...questionPT,
    ...surveyPT,
  },
};
