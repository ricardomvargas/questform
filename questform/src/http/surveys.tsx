import axios from 'axios';

import { surveysGet, surveysGetById } from './enpoints';

// #TODO: Add error handler
export const getAllSurveys = (page = '0', title = '', status = '', limit = '') =>
  axios.get(`${surveysGet}?page=${page}&title=${title}&status=${status}&limit=${limit}`);

export const getLastSurveys = (limit = '5') => axios.get(`${surveysGet}?limit=${limit}&order=desc`);

export const getSurveyById = (idForm: number) => axios.get(`${surveysGetById}?idform=${idForm}`);
