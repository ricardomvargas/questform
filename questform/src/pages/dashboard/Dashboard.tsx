import React, { useEffect, useState } from 'react';
import { useIntl } from 'react-intl';

import { TSurvey } from '../../types/Entities/Survey';
import { TServerResponseList } from '../../types/Http';
import { ListItem } from '../../types/components/DataTable';

import DataTable from '../../components/DataTable/DataTable';
import DashBoardCard from '../../components/DashboardCard/DashboardCard';
import IconButton from '../../components/IconButton/IconButton';

import { getLastSurveys } from '../../http/surveys';
import { surveysEditRoute } from '../../util/routes';

import { PURPLE, GREEN, BLUE, RED, ARROW_LEFT, ARROW_RIGHT } from '../../util/constants';
import { BuildDataTableLine } from '../../util/util';

const moreOptions = [
  {
    title: 'View',
    url: surveysEditRoute,
  },
];

const Dashboard = () => {
  const intl = useIntl();
  const [surveys, setSurveys] = useState<{
    list: ListItem[];
    totalItens: number;
  }>({ list: [], totalItens: 0 });

  useEffect(() => {
    const fetchData = async () => {
      const surveyResponse = await getLastSurveys();
      const { data } = surveyResponse;
      const { list, total } = (data as TServerResponseList) || {};
      const surveys: ListItem[] =
        total > 0 ? list.map((survey: TSurvey) => BuildDataTableLine(survey, intl)) : [];

      setSurveys({ list: surveys, totalItens: total });
    };
    fetchData();
    /** #TODO: Entender por que o eslint reclama de n√£o ter a dependecia do intl
     * nesse useEffect. */
    //eslint-disable-next-line
  }, []);

  return (
    <div className="app-body-content-no-margin">
      <div className="page-header">
        <h1>{intl.formatMessage({ id: 'dashboardTitle' })}</h1>
      </div>
      <div className="app-body-content">
        <section className="cards-section">
          <div>
            <IconButton
              buttonType={ARROW_LEFT}
              title={intl.formatMessage({ id: 'dashboardPreviouslyWeek' })}
              onClickAction={() => console.log('arrow right left')}
            />
            <h2>10/04/2021 to 10/08/2021</h2>
            <IconButton
              buttonType={ARROW_RIGHT}
              title={intl.formatMessage({ id: 'dashboardNextWeek' })}
              onClickAction={() => console.log('arrow right click')}
            />
          </div>
          <hr />
          <div>
            <DashBoardCard skin={PURPLE} title={'loren inpsum'} content={'0'} />
            <DashBoardCard skin={GREEN} title={'loren inpsum'} content={'0'} />
            <DashBoardCard skin={BLUE} title={'loren inpsum'} content={'0'} />
            <DashBoardCard skin={RED} title={'loren inpsum'} content={'0'} />
          </div>
        </section>
        <section className="last-surveys-section">
          <h4>{intl.formatMessage({ id: 'dashboardLastSurveysAnswered' })}</h4>
          <DataTable
            dataList={surveys.list}
            totalItens={surveys.totalItens}
            hideTitle
            moreOptions={moreOptions}
          />
        </section>
      </div>
    </div>
  );
};

export default Dashboard;
