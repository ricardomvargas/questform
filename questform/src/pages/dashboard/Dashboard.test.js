/**
 * @jest-environment jsdom
 */
import React from 'react';
import { IntlProvider } from 'react-intl';
import '@testing-library/jest-dom';
import { render, act } from '@testing-library/react';
import axios from 'axios';

import messages from '../../intl/messages';
import Dashboard from './Dashboard';

import { getLastSurveys } from '../../http/surveys';

jest.mock('axios');

const Wraper = ({ children }) => (
  <IntlProvider locale="en" messages={messages.en}>
    {children}
  </IntlProvider>
);

test('Check if match with snapshot', async () => {
  axios.get.mockResolvedValue({
    data: {
      list: [
        {
          idSurvey: 18,
          title: 'Sed sodales congue consequat',
          idSurveyStatus: 2,
          publicTitle:
            'In tempor elit quis viverra tristique. Nullam auctor ultricies condimentum. Donec venenatis vel neque in porta. Mauris lacinia vitae nisl vitae tristique.',
          publicDescription:
            'Nam mattis, neque eget commodo viverra, neque dolor bibendum lectus, finibus dignissim orci purus quis arcu. Praesent mattis pellentesque arcu. Vivamus mattis velit vitae risus bibendum, sed gravida arcu commodo. Proin eget sem massa. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae;',
          consclusionDescription:
            'Quisque dapibus nunc arcu, non egestas arcu volutpat at. Phasellus vestibulum posuere elit sit amet consectetur. Nulla congue vel ipsum vel mattis. Proin laoreet ante vitae orci vestibulum, eu faucibus mauris imperdiet. Etiam ultricies ligula quis maximus blandit. Nulla ut nunc eu lectus euismod scelerisque. Donec euismod varius nibh quis facilisis.',
          createDate: '2021-03-21 16:20:51',
          active: 1,
          totalQuestions: 0,
          results: 18,
        },
      ],
      total: 18,
    },
  });

  const dasboard = render(<Wraper children={<Dashboard />} />);
  /**
   * About act:
   * If the code bellow will not be wraped in the act(), React will be giving a warning in the console saying that the code should
   * be wrapped with act. This happen because after the promisse be resolved, something in the page will be updated and the test
   * need to tell React about that, so the request must be wrapped in act();
   */
  await act(() => getLastSurveys());
  expect(dasboard).toMatchSnapshot();
});
