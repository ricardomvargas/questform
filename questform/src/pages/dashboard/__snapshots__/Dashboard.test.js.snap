// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app-body-content-no-margin"
      >
        <div
          class="page-header"
        >
          <h1>
            My Dashboard
          </h1>
        </div>
        <div
          class="app-body-content"
        >
          <section
            class="cards-section"
          >
            <div>
              <a
                title="Previously week"
              >
                <svg
                  enable-background="new 0 0 15.531 15.5"
                  height="15.5px"
                  id="Layer_1"
                  version="1.1"
                  viewBox="0 0 15.531 15.5"
                  width="15.531px"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g>
                    <g
                      id="btn-add_4_"
                    >
                      <circle
                        cx="7.764"
                        cy="7.762"
                        fill="#CACBCC"
                        r="7.472"
                        stroke="#878787"
                        stroke-miterlimit="10"
                        stroke-width="0.5337"
                      />
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="SVGID_1_"
                        x1="7.835"
                        x2="7.835"
                        y1="1.9565"
                        y2="13.7109"
                      >
                        <stop
                          offset="0.4"
                          style="stop-color: rgb(202, 203, 204);"
                        />
                        <stop
                          offset="1"
                          style="stop-color: rgb(135, 135, 135);"
                        />
                      </lineargradient>
                      <circle
                        cx="7.835"
                        cy="7.833"
                        fill="url(#SVGID_1_)"
                        r="5.877"
                      />
                    </g>
                    <polygon
                      fill="#FFFFFF"
                      points="6.706,7.374 9.625,5.689 9.625,4.534 4.706,7.374 9.625,10.214 9.625,9.059 	"
                    />
                  </g>
                </svg>
              </a>
              <h2>
                10/04/2021 to 10/08/2021
              </h2>
              <a
                title="Next week"
              >
                <svg
                  enable-background="new 0 0 15.469 15.5"
                  height="15.5px"
                  id="Layer_1"
                  version="1.1"
                  viewBox="0 0 15.469 15.5"
                  width="15.469px"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g>
                    <g
                      id="btn-add_1_"
                    >
                      <circle
                        cx="7.733"
                        cy="7.762"
                        fill="#CACBCC"
                        r="7.472"
                        stroke="#878787"
                        stroke-miterlimit="10"
                        stroke-width="0.5337"
                      />
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="SVGID_1_"
                        x1="7.8037"
                        x2="7.8037"
                        y1="1.9565"
                        y2="13.7104"
                      >
                        <stop
                          offset="0.4"
                          style="stop-color: rgb(202, 203, 204);"
                        />
                        <stop
                          offset="1"
                          style="stop-color: rgb(135, 135, 135);"
                        />
                      </lineargradient>
                      <circle
                        cx="7.804"
                        cy="7.833"
                        fill="url(#SVGID_1_)"
                        r="5.877"
                      />
                    </g>
                    <polygon
                      fill="#FFFFFF"
                      points="8.138,7.374 5.219,9.059 5.219,10.214 10.138,7.374 5.219,4.534 5.219,5.689 	"
                    />
                  </g>
                </svg>
              </a>
            </div>
            <hr />
            <div>
              <div
                class="purple-card roboto-condensed"
              >
                <h4
                  class="roboto-condensed-bold"
                >
                  loren inpsum
                </h4>
                <p>
                  0
                </p>
              </div>
              <div
                class="green-card roboto-condensed"
              >
                <h4
                  class="roboto-condensed-bold"
                >
                  loren inpsum
                </h4>
                <p>
                  0
                </p>
              </div>
              <div
                class="blue-card roboto-condensed"
              >
                <h4
                  class="roboto-condensed-bold"
                >
                  loren inpsum
                </h4>
                <p>
                  0
                </p>
              </div>
              <div
                class="red-card roboto-condensed"
              >
                <h4
                  class="roboto-condensed-bold"
                >
                  loren inpsum
                </h4>
                <p>
                  0
                </p>
              </div>
            </div>
          </section>
          <section
            class="last-surveys-section"
          >
            <h4>
              Last survey's aswered
            </h4>
            <div
              class="data-table"
            >
              <ul
                class="data-table-content-no-title"
              >
                <li
                  class="data-item"
                >
                  <div
                    class="item-status"
                  >
                    <span
                      class="online"
                    />
                  </div>
                  <div
                    class="item-text"
                  >
                    <span>
                      18 - Sed sodales congue consequat
                    </span>
                    <span
                      class="sub-text"
                    >
                      Total questions: 0
                    </span>
                  </div>
                  <div
                    class="more-options"
                  >
                    <div
                      class="options"
                    >
                      <a>
                        <svg
                          class="icon-button icon-button-dark "
                          height="512pt"
                          viewBox="0 -192 512 512"
                          width="512pt"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m320 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                          />
                          <path
                            d="m128 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                          />
                          <path
                            d="m512 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                          />
                        </svg>
                      </a>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app-body-content-no-margin"
    >
      <div
        class="page-header"
      >
        <h1>
          My Dashboard
        </h1>
      </div>
      <div
        class="app-body-content"
      >
        <section
          class="cards-section"
        >
          <div>
            <a
              title="Previously week"
            >
              <svg
                enable-background="new 0 0 15.531 15.5"
                height="15.5px"
                id="Layer_1"
                version="1.1"
                viewBox="0 0 15.531 15.5"
                width="15.531px"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g>
                  <g
                    id="btn-add_4_"
                  >
                    <circle
                      cx="7.764"
                      cy="7.762"
                      fill="#CACBCC"
                      r="7.472"
                      stroke="#878787"
                      stroke-miterlimit="10"
                      stroke-width="0.5337"
                    />
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="SVGID_1_"
                      x1="7.835"
                      x2="7.835"
                      y1="1.9565"
                      y2="13.7109"
                    >
                      <stop
                        offset="0.4"
                        style="stop-color: rgb(202, 203, 204);"
                      />
                      <stop
                        offset="1"
                        style="stop-color: rgb(135, 135, 135);"
                      />
                    </lineargradient>
                    <circle
                      cx="7.835"
                      cy="7.833"
                      fill="url(#SVGID_1_)"
                      r="5.877"
                    />
                  </g>
                  <polygon
                    fill="#FFFFFF"
                    points="6.706,7.374 9.625,5.689 9.625,4.534 4.706,7.374 9.625,10.214 9.625,9.059 	"
                  />
                </g>
              </svg>
            </a>
            <h2>
              10/04/2021 to 10/08/2021
            </h2>
            <a
              title="Next week"
            >
              <svg
                enable-background="new 0 0 15.469 15.5"
                height="15.5px"
                id="Layer_1"
                version="1.1"
                viewBox="0 0 15.469 15.5"
                width="15.469px"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g>
                  <g
                    id="btn-add_1_"
                  >
                    <circle
                      cx="7.733"
                      cy="7.762"
                      fill="#CACBCC"
                      r="7.472"
                      stroke="#878787"
                      stroke-miterlimit="10"
                      stroke-width="0.5337"
                    />
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="SVGID_1_"
                      x1="7.8037"
                      x2="7.8037"
                      y1="1.9565"
                      y2="13.7104"
                    >
                      <stop
                        offset="0.4"
                        style="stop-color: rgb(202, 203, 204);"
                      />
                      <stop
                        offset="1"
                        style="stop-color: rgb(135, 135, 135);"
                      />
                    </lineargradient>
                    <circle
                      cx="7.804"
                      cy="7.833"
                      fill="url(#SVGID_1_)"
                      r="5.877"
                    />
                  </g>
                  <polygon
                    fill="#FFFFFF"
                    points="8.138,7.374 5.219,9.059 5.219,10.214 10.138,7.374 5.219,4.534 5.219,5.689 	"
                  />
                </g>
              </svg>
            </a>
          </div>
          <hr />
          <div>
            <div
              class="purple-card roboto-condensed"
            >
              <h4
                class="roboto-condensed-bold"
              >
                loren inpsum
              </h4>
              <p>
                0
              </p>
            </div>
            <div
              class="green-card roboto-condensed"
            >
              <h4
                class="roboto-condensed-bold"
              >
                loren inpsum
              </h4>
              <p>
                0
              </p>
            </div>
            <div
              class="blue-card roboto-condensed"
            >
              <h4
                class="roboto-condensed-bold"
              >
                loren inpsum
              </h4>
              <p>
                0
              </p>
            </div>
            <div
              class="red-card roboto-condensed"
            >
              <h4
                class="roboto-condensed-bold"
              >
                loren inpsum
              </h4>
              <p>
                0
              </p>
            </div>
          </div>
        </section>
        <section
          class="last-surveys-section"
        >
          <h4>
            Last survey's aswered
          </h4>
          <div
            class="data-table"
          >
            <ul
              class="data-table-content-no-title"
            >
              <li
                class="data-item"
              >
                <div
                  class="item-status"
                >
                  <span
                    class="online"
                  />
                </div>
                <div
                  class="item-text"
                >
                  <span>
                    18 - Sed sodales congue consequat
                  </span>
                  <span
                    class="sub-text"
                  >
                    Total questions: 0
                  </span>
                </div>
                <div
                  class="more-options"
                >
                  <div
                    class="options"
                  >
                    <a>
                      <svg
                        class="icon-button icon-button-dark "
                        height="512pt"
                        viewBox="0 -192 512 512"
                        width="512pt"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m320 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                        />
                        <path
                          d="m128 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                        />
                        <path
                          d="m512 64c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
